2024-07-11 12:36:41.414263: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2024-07-11 12:36:41.414467: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2024-07-11 12:36:49,947 | app.py:165 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2024-07-11 12:36:50,050 | app.py:179 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2024-07-11 12:36:50,050 | server.py:89 | Initializing global parameters
INFO flwr 2024-07-11 12:36:50,050 | server.py:276 | Requesting initial parameters from one random client
INFO flwr 2024-07-11 12:36:51,135 | server.py:280 | Received initial parameters from one random client
INFO flwr 2024-07-11 12:36:51,136 | server.py:91 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:51,152 | server.py:98 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.4033333333333333}
INFO flwr 2024-07-11 12:36:51,153 | server.py:104 | FL starting
DEBUG flwr 2024-07-11 12:36:51,427 | server.py:227 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2024-07-11 12:36:51,456 | server.py:241 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2024-07-11 12:36:51,459 | fedavg.py:242 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:51,468 | server.py:131 | fit progress: (1, 0.3799178573249177, {'accuracy': 0.8266666666666667}, 0.3155070000000002)
DEBUG flwr 2024-07-11 12:36:51,469 | server.py:178 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:36:51,495 | server.py:192 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2024-07-11 12:36:51,496 | fedavg.py:273 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2024-07-11 12:36:51,497 | server.py:227 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:36:51,523 | server.py:241 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:51,532 | server.py:131 | fit progress: (2, 0.3820840991250383, {'accuracy': 0.8266666666666667}, 0.37961510000000054)
DEBUG flwr 2024-07-11 12:36:51,532 | server.py:178 | evaluate_round 2: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:36:51,557 | server.py:192 | evaluate_round 2 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:36:51,557 | server.py:227 | fit_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:36:51,585 | server.py:241 | fit_round 3 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:51,596 | server.py:131 | fit progress: (3, 0.38331702537060247, {'accuracy': 0.8266666666666667}, 0.44380160000000046)
DEBUG flwr 2024-07-11 12:36:51,597 | server.py:178 | evaluate_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:36:51,618 | server.py:192 | evaluate_round 3 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:36:51,619 | server.py:227 | fit_round 4: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:36:51,651 | server.py:241 | fit_round 4 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:51,663 | server.py:131 | fit progress: (4, 0.3837530816798102, {'accuracy': 0.82}, 0.5106029999999997)
DEBUG flwr 2024-07-11 12:36:51,664 | server.py:178 | evaluate_round 4: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:36:51,690 | server.py:192 | evaluate_round 4 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:36:51,691 | server.py:227 | fit_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:36:51,726 | server.py:241 | fit_round 5 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:51,737 | server.py:131 | fit progress: (5, 0.38375233617145776, {'accuracy': 0.82}, 0.5842324000000012)
DEBUG flwr 2024-07-11 12:36:51,738 | server.py:178 | evaluate_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:36:51,766 | server.py:192 | evaluate_round 5 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:36:51,766 | server.py:227 | fit_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:36:51,799 | server.py:241 | fit_round 6 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:51,810 | server.py:131 | fit progress: (6, 0.3837510510625477, {'accuracy': 0.82}, 0.6580238999999999)
DEBUG flwr 2024-07-11 12:36:51,811 | server.py:178 | evaluate_round 6: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:51,845 | server.py:192 | evaluate_round 6 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:36:51,846 | server.py:227 | fit_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:51,887 | server.py:241 | fit_round 7 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:51,900 | server.py:131 | fit progress: (7, 0.38307474202379654, {'accuracy': 0.82}, 0.7479031999999997)
DEBUG flwr 2024-07-11 12:36:51,901 | server.py:178 | evaluate_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:51,939 | server.py:192 | evaluate_round 7 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:36:51,940 | server.py:227 | fit_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:51,987 | server.py:241 | fit_round 8 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:52,006 | server.py:131 | fit progress: (8, 0.38307601661984736, {'accuracy': 0.82}, 0.8532817999999995)
DEBUG flwr 2024-07-11 12:36:52,006 | server.py:178 | evaluate_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:52,038 | server.py:192 | evaluate_round 8 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:36:52,038 | server.py:227 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:52,079 | server.py:241 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:52,091 | server.py:131 | fit progress: (9, 0.38307720072791646, {'accuracy': 0.82}, 0.9382045000000012)
DEBUG flwr 2024-07-11 12:36:52,091 | server.py:178 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:52,121 | server.py:192 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:36:52,122 | server.py:227 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:52,156 | server.py:241 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:52,168 | server.py:131 | fit progress: (10, 0.3830772243742687, {'accuracy': 0.82}, 1.0150454)
DEBUG flwr 2024-07-11 12:36:52,168 | server.py:178 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:52,201 | server.py:192 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2024-07-11 12:36:52,201 | server.py:153 | FL finished in 1.0484714999999998
INFO flwr 2024-07-11 12:36:52,202 | app.py:225 | app_fit: losses_distributed [(1, 0.4071575452884038), (2, 0.4050816426674525), (3, 0.40445627172787985), (4, 0.41006146525514536), (5, 0.41006176995820015), (6, 0.4144209155014583), (7, 0.41430799961090087), (8, 0.41430791105542863), (9, 0.4143078633717128), (10, 0.4143078646489552)]
INFO flwr 2024-07-11 12:36:52,202 | app.py:226 | app_fit: metrics_distributed_fit {}
INFO flwr 2024-07-11 12:36:52,202 | app.py:227 | app_fit: metrics_distributed {}
INFO flwr 2024-07-11 12:36:52,202 | app.py:228 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.3799178573249177), (2, 0.3820840991250383), (3, 0.38331702537060247), (4, 0.3837530816798102), (5, 0.38375233617145776), (6, 0.3837510510625477), (7, 0.38307474202379654), (8, 0.38307601661984736), (9, 0.38307720072791646), (10, 0.3830772243742687)]
INFO flwr 2024-07-11 12:36:52,203 | app.py:229 | app_fit: metrics_centralized {'accuracy': [(0, 0.4033333333333333), (1, 0.8266666666666667), (2, 0.8266666666666667), (3, 0.8266666666666667), (4, 0.82), (5, 0.82), (6, 0.82), (7, 0.82), (8, 0.82), (9, 0.82), (10, 0.82)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.89042991,  0.97628326,  0.69691614,  0.49840157,  0.27666838,
         0.0601275 , -0.01396716, -0.0495638 ]]), array([0.06992467])]
[array([[-1.89042991,  0.97628326,  0.69691614,  0.49840157,  0.27666838,
         0.0601275 , -0.01396716, -0.0495638 ]]), array([0.06992467])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.89231707,  0.93082424,  0.74134926,  0.47721945,  0.23119038,
         0.09512688, -0.05603622,  0.02566023]]), array([-0.05797295])]
[array([[-1.89231707,  0.93082424,  0.74134926,  0.47721945,  0.23119038,
         0.09512688, -0.05603622,  0.02566023]]), array([-0.05797295])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.93880126,  0.98756626,  0.76291966,  0.48557214,  0.18439234,
         0.09664304, -0.04029909, -0.01490251]]), array([-0.04441195])]
[array([[-1.93880126,  0.98756626,  0.76291966,  0.48557214,  0.18439234,
         0.09664304, -0.04029909, -0.01490251]]), array([-0.04441195])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96888635,  0.97160306,  0.77039257,  0.4950053 ,  0.19457991,
         0.08238023, -0.04317733,  0.00809519]]), array([-0.06104604])]
[array([[-1.96888635,  0.97160306,  0.77039257,  0.4950053 ,  0.19457991,
         0.08238023, -0.04317733,  0.00809519]]), array([-0.06104604])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96842319,  0.9715271 ,  0.77050783,  0.49497931,  0.19472739,
         0.08257251, -0.04314827,  0.00840269]]), array([-0.06216542])]
[array([[-1.96842319,  0.9715271 ,  0.77050783,  0.49497931,  0.19472739,
         0.08257251, -0.04314827,  0.00840269]]), array([-0.06216542])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96843858,  0.97149208,  0.77047166,  0.49497351,  0.1947173 ,
         0.08254734, -0.04316462,  0.00836671]]), array([-0.06196258])]
[array([[-1.96843858,  0.97149208,  0.77047166,  0.49497351,  0.1947173 ,
         0.08254734, -0.04316462,  0.00836671]]), array([-0.06196258])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.93135765,  0.95117831,  0.7698308 ,  0.46039182,  0.20170037,
         0.08055134, -0.03407025,  0.02263126]]), array([-0.05739444])]
[array([[-1.93135765,  0.95117831,  0.7698308 ,  0.46039182,  0.20170037,
         0.08055134, -0.03407025,  0.02263126]]), array([-0.05739444])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.93164283,  0.95101292,  0.76992641,  0.45980301,  0.20150719,
         0.08026363, -0.03410754,  0.0228255 ]]), array([-0.05607634])]
[array([[-1.93164283,  0.95101292,  0.76992641,  0.45980301,  0.20150719,
         0.08026363, -0.03410754,  0.0228255 ]]), array([-0.05607634])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.93165787,  0.9510289 ,  0.7699563 ,  0.45980841,  0.20149971,
         0.08028067, -0.03409834,  0.0228477 ]]), array([-0.05617399])]
[array([[-1.93165787,  0.9510289 ,  0.7699563 ,  0.45980841,  0.20149971,
         0.08028067, -0.03409834,  0.0228477 ]]), array([-0.05617399])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.93165632,  0.95102769,  0.76995593,  0.45980989,  0.20149913,
         0.08028086, -0.03409841,  0.02284904]]), array([-0.05617566])]
[array([[-1.93165632,  0.95102769,  0.76995593,  0.45980989,  0.20149913,
         0.08028086, -0.03409841,  0.02284904]]), array([-0.05617566])]
>>>>>>>>>>>>>run time:
0.5625s
